src/index.js

```diff
@@ -0,0 +1,245 @@
+ * @param {object?} options.repoOptions - Options that are passed to migrations, that can use them to correctly construct datastore. Options are same like for IPFSRepo.
+ * @param {function?} options.onProgress - Callback which will be called after each executed migration to report progress
+ * @param {boolean?} options.isDryRun - Allows to simulate the execution of the migrations without any effect.
+ * @param {array?} options.migrations - Array of migrations to migrate. If undefined, the bundled migrations are used. Mainly for testing purpose.
+ * @returns {Promise<void>}
+ */
+async function migrate (path, { toVersion, ignoreLock = false, repoOptions, onProgress, isDryRun = false, migrations }) {
+  migrations = migrations || defaultMigrations
+
+  if (!path) {
+    throw new errors.RequiredParameterError('Path argument is required!')
+  }
+
+  if (!(await repoInit.isRepoInitialized(path))) {
+    throw new errors.NotInitializedRepoError(`Repo in path ${path} is not initialized!`)
+  }
+
+  if (toVersion && (!Number.isInteger(toVersion) || toVersion <= 0)) {
+    throw new errors.InvalidValueError('Version has to be positive integer!')
+  }
+  toVersion = toVersion || getLatestMigrationVersion(migrations)
+
+  if (toVersion > getLatestMigrationVersion(migrations)) {
+    throw new errors.InvalidValueError('The ipfs-repo-migrations package does not have migration for version: ' + toVersion)
+  }
+
+  const currentVersion = await repoVersion.getVersion(path)
+
+  if (currentVersion === toVersion) {
+    log('Nothing to migrate, skipping migrations.')
+    return
+  }
+
+  if (currentVersion > toVersion) {
+    log(`Current repo's version (${currentVersion}) is higher then toVersion (${toVersion}), nothing to migrate.`)
+    return
+  }
+
+  let lock
+  if (!isDryRun && !ignoreLock) lock = await repoLock.lock(currentVersion, path)
+
+  try {
+    let counter = 0
+    let totalMigrations = toVersion - currentVersion
+    for (let migration of migrations) {
+      if (toVersion !== undefined && migration.version > toVersion) {
+        break
+      }
+      if (migration.version <= currentVersion) {
+        continue
+      }
+
+      counter++
+      log(`Migrating version ${migration.version}`)
+      try {
+        if (!isDryRun) await migration.migrate(path, repoOptions, isBrowser)
+      } catch (e) {
+        const lastSuccessfullyMigratedVersion = migration.version - 1
+        log(`An exception was raised during execution of migration. Setting the repo's version to last successfully migrated version: ${lastSuccessfullyMigratedVersion}`)
+        await repoVersion.setVersion(path, lastSuccessfullyMigratedVersion)
+
+        e.message = `During migration to version ${migration.version} exception was raised: ${e.message}`
+        throw e
+      }
+
+      typeof onProgress === 'function' && onProgress(migration, counter, totalMigrations) // Reports on migration process
```

Comment:
Nit, personally I'd default this to a noop function (`onProgress = onProgress || (() => {})`) instead of checking every time and using &&. It makes this line a little more readable and less confusing to more junior programmers.